src/components/Error/ErrorFallback.tsx

```diff
@@ -1,16 +1,47 @@
-import { FC } from "react";
+import { useCallback } from "react";
+import { faRefresh } from "@fortawesome/free-solid-svg-icons";
+import { match } from "ts-pattern";
+import { useNavigate } from "react-router-dom";
 import { FallbackProps } from "react-error-boundary";
 import { Stack } from "@deskpro/app-sdk";
+import { GitLabError } from "../../services/gitlab";
 import { ErrorBlock } from "./ErrorBlock";
-import { Container } from "../common";
+import { Container, Button } from "../common";
+import type { FC } from "react";
 
 type Props = Omit<FallbackProps, "error"> & {
-    error: Error,
+    error: Error|GitLabError,
 };
 
-const ErrorFallback: FC<Props> = ({ /*resetErrorBoundary, */error }) => {
-    const message = "There was an error!";
-    const nativeErrorMessage = error.message;
+const ErrorFallback: FC<Props> = ({ resetErrorBoundary, error }) => {
+    const navigate = useNavigate();
+
+    let message = "There was an error!";
+    let button = null;
+    const nativeErrorMessage = error;
+
+    const toLogin = useCallback(() => {
+        resetErrorBoundary();
+        navigate("/login");
+    }, [navigate, resetErrorBoundary]);
+
+    if (error instanceof GitLabError) {
+        const { status } = error;
+
+        match(status)
```

Comment:
nitpicking, but if this is not an expression (i.e. setting variables inside of each case) then you're better off using a `switch` block instead of `match()`

Reply:
Thanks for the comment, Ash. In the future, I'll do it as you say